timestamp-api/src/main/java/com/palantir/timestamp/InMemoryTimestampService.java

```diff
@@ -54,4 +54,9 @@ public void fastForwardTimestamp(long currentTimestamp) {
     public String ping() {
         return PING_RESPONSE;
     }
+
+    @Override
+    public long getUpperLimitTimestampToHandOutInclusive() {
```

Comment:
if going to be unimplemented, why change superinterface out of curiosity?

Reply:
Just because I needed a single interface to use with @AutoDelegate for AsyncInitializeableInMemoryTimestampService

Reply:
I agree with felix - this interface just isn't implemented!

Reply:
like, if you need something for use in testing, then by all means add some small piece of test-only code which encapsulate the hacks you need to write your test - but we shouldn't compromise our exported code (by e.g. incorrectly implementing interfaces) in order to make testing easier - it implies that the decomp is incorrect.

The Atlas code should make sense as something someone can read, and we shouldn't have to say 'oh this class doesn't actually make sense as anything you'd want to run, but it's easier to write this test (that no longer exists because someone refactored it away) if it works this way'.